services/validation/README.md

```diff
@@ -1,15 +1,128 @@
+
+## Usage as a script
 
 * `npm install -g sourcify-validation`
-* `sourcify-validation path/to/file1 path/to/file2 path/to/dir path/to/zip`
-    * the script will scan through all the provided files, directories and zips searching for metadata and the belonging source files
-* If everything goes well, proceed here: [Sourcify](https://verification.komputing.org/)
+* `sourcify-validation my/repository`
+
+You can specify a sequence of files, directories and even zip files.
+The script will scan through all the provided files, searching for metadata and the belonging source files.
+
+If everything goes well, you can ask it to provide a "standard-json input file"
+that you just need to send to the compiler, etherscan, or whatever service you use
+for verification.
+
+Verification is also achievable through the use of [Sourcify](https://verification.komputing.org/). All you need to do is upload the files
+you had provided to this script.
+
+## Options
+* -j, --prepare-json <path:name>
+  * Output only a JSON object to be used as standard-json input to solc.
+  * The expected argument ought to be of the form path:name
+  * E.g. `sourcify-validation --prepare-json foo/bar/Contract.sol:ContractName path/to/dir`
+  * Omitting either the path or the name is tolerated if unambiguous.
+* -S, --no-settings
+  * Omit the original settings when preparing a standard-json.
+  * Only available with the --prepare-json option.
+* -p, --pretty
+  * Output the standard json in a pretty format.
+  * Only available with the --prepare-json option.
+* -h, --help
+  * Output the help message.
+* -v, --version
+  * Output the current version.
+
+## Example - Piping the standard-json input to the Solidity compiler
+Consider the case of a metadata.json file specifying a single source file (1_Storage.sol), both located inside the current directory.
+
+The file `metadata.json` is the essential part: it contains the compilation parameters. During compilation, you get it using `solc --metadata ...`
+
+Since Sourcify Validation will only consider the relevant files, you can run the command as:
+### Command
+* `sourcify-validation .`
+
+### Output
+```
+Storage (browser/1_Storage.sol):
+  Success!
+  Compiled with Solidity 0.6.6
+  https://solc-bin.ethereum.org/wasm/soljson-v0.6.6+commit.6c089d02.js
+  https://solc-bin.ethereum.org/linux-amd64/solc-linux-amd64-v0.6.6+commit.6c089d02
+  To recompile, use: `sourcify-validation --prepare-json browser/1_Storage.sol:Storage /path/to/dir | solc --standard-json`
+```
+
+If you don't have the required version of the compiler, run:
+* `wget https://solc-bin.ethereum.org/linux-amd64/solc-linux-amd64-v0.6.6+commit.6c089d02 -O solc-0.6.6`
 
-## Example - multiple sources
+Then run:
+* `sourcify-validation -j browser/1_storage.sol:Storage . | solc-0.6.6 --standard-json`
+
+If you use `jq`, the output will have a nice format:
```

Comment:
we can omit that with the pretty-printing option.

Reply:
Actually the pretty-printing option of sourcify-validation only affects its output. Here the jq can be used to pretty-print the output of solc.